esp/services/ws_ecl/ws_ecl_service.cpp

```diff
@@ -1993,20 +1993,18 @@ int CWsEclBinding::submitWsEclWorkunit(IEspContext & context, WsEclWuInfo &wsinf
     StringAttr wuid(workunit->queryWuid());  // NB queryWuid() not valid after workunit,clear()
 
     bool noTimeout = false;
-    if (httpreq)
+
+    Owned<ISpan> clientSpan;
+    ISpan * activeSpan = context.queryActiveSpan();
+    if (activeSpan)
     {
-        StringBuffer globalId, callerId;
-        wsecl->getHttpGlobalIdHeader(httpreq, globalId);
-        wsecl->getHttpCallerIdHeader(httpreq, callerId);
-        if (globalId.length())
-        {
-            workunit->setDebugValue("GlobalId", globalId.str(), true);
+        clientSpan.setown(activeSpan->createClientSpan("wsecl"));
```

Comment:
@afishbeck I think this code was wrong before - I didn't pick up the local id (or global id?) assigned if none was provided by the http request.

Reply:
This probably wasn't tested much when it came to global-ids but is a good fix, and we definitely want it working for trace/span support.

Reply:
@ghalliday but a more descriptive name for the span might be good.  Something like "wsecl/workunit" or similar.  We should probably try and be consistent in general format.  I stole that particular format from the opentelemetry demo which has one of it's services written in c++.  https://github.com/open-telemetry/opentelemetry-demo/blob/fa529260273fa53b4e018daa73f30b1d30b88b3a/src/currencyservice/src/server.cpp#L112

That service seems to use the form `"<servicename>/<operation>"`.  For our client spans I think something like may work.